packages/p2p/src/components/my-ads/form-ads.jsx

```diff
@@ -12,9 +14,21 @@ import { buy_sell } from '../../constants/buy-sell';
 const FormAds = observer(() => {
     const { general_store, my_ads_store } = useStores();
     const { currency, local_currency_config } = general_store.client;
+    const [is_api_error_modal_visible, setIsApiErrorModalVisible] = React.useState(false);
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    React.useEffect(() => my_ads_store.getAdvertiserInfo(), []);
+    React.useEffect(() => {
+        my_ads_store.getAdvertiserInfo();
+        return () => my_ads_store.setApiErrorMessage('');
+    }, []);
+
+    React.useEffect(() => {
+        return reaction(
+            () => my_ads_store.api_error_message,
+            () => toggleApiErrorModal(!!my_ads_store.api_error_message)
+        );
+    }, []);
```

Comment:
- You can combine these into a single effect. 
    - `reaction` returns a disposer function that can be called in the clean up effect.
- There is no need to abstract `setIsApiErrorModalVisible` into `toggleApiErrorModal` (they're the same), you can just call it directly (line 27 and 275).

```JS
React.useEffect(() => {
    my_ads_store.getAdvertiserInfo();
    
    const disposeApiErrorReaction = reaction(
        () => my_ads_store.api_error_message,
        () => setIsApiErrorModalVisible(!!my_ads_store.api_error_message)
    );

    return () => {
        disposeApiErrorReaction();
        my_ads_store.setApiErrorMessage('');
    }
}, []);
```